core/domain/opportunity_services.py

```diff
@@ -0,0 +1,374 @@
+                opportunity_summary.incomplete_translation_languages),
+            translation_counts=opportunity_summary.translation_counts,
+            need_voice_artist_in_languages=(
+                opportunity_summary.need_voice_artist_in_languages),
+            assigned_voice_artist_in_languages=(
+                opportunity_summary.assigned_voice_artist_in_languages)
+        )
+
+        exploration_opportunity_summary_model_list.append(model)
+
+    opportunity_models.ExplorationOpportunitySummaryModel.put_multi(
+        exploration_opportunity_summary_model_list)
+
+
+def add_new_exploration_opportunities(story, exp_ids):
+    """Adds new exploration opportunity into the model.
+
+    Args:
+        story: Story. A story object.
+        exp_ids: list(str). A list of exploration ids for which new
+            opportunities are to be created.
+    """
+    topic = topic_services.get_topic_by_id(story.corresponding_topic_id)
+    explorations = exp_fetchers.get_multiple_explorations_by_id(exp_ids)
+
+    exploration_opportunity_summary_list = []
+    for exp_id, exploration in explorations.iteritems():
+        node = story.story_contents.get_node_with_corresponding_exp_id(exp_id)
+        if node is None:
+            continue
+
+        audio_language_codes = set([
+            language['id'] for language in constants.SUPPORTED_AUDIO_LANGUAGES])
+
+        complete_translation_languages = set(
+            exploration.get_languages_with_complete_translation())
+
+        incomplete_translation_languages = (
+            audio_language_codes - complete_translation_languages)
+        need_voice_artist_in_languages = complete_translation_languages
+
+        if exploration.language_code in incomplete_translation_languages:
+            # Removing exploration language from incomplete translation
+            # languages list as exploration does not need any translation in
+            # it's own language.
+            incomplete_translation_languages.discard(exploration.language_code)
+            # Adding exploration language to voiceover required languages
+            # list as exploration can be voiceovered in it's own language.
+            need_voice_artist_in_languages.add(exploration.language_code)
+
+        content_count = exploration.get_content_count()
+        translation_counts = exploration.get_translation_counts()
+
+        exploration_opportunity_summary = (
+            opportunity_domain.ExplorationOpportunitySummary(
+                exp_id, topic.id, topic.name, story.id, story.title, node.title,
+                content_count, list(incomplete_translation_languages),
+                translation_counts, list(need_voice_artist_in_languages), []))
+
+        exploration_opportunity_summary_list.append(
+            exploration_opportunity_summary)
+
+    _save_multi_exploration_opportunity_summary(
+        exploration_opportunity_summary_list)
+
+
+def update_exploration_opportunity_with_new_exploration(
```

Comment:
This name doesn't feel right. There is no new exploration right? Its just a new exploration version?

Reply:
I've updated the method name: `update_opportunity_with_updated_exploration`. (Done!)